nntrainer/graph/network_graph.cpp

```diff
@@ -329,8 +329,10 @@ void NetworkGraph::applyGradients(
   }
 }
 
-sharedConstTensors NetworkGraph::forwarding(bool training) const {
-  for (auto iter = cbegin(); iter != cend(); iter++) {
+sharedConstTensors
+NetworkGraph::forwarding(bool training,
+                         std::function<bool(void *userdata)> stop_cb) const {
+  for (auto iter = cbegin(); iter != cend() && !stop_cb(nullptr); iter++) {
```

Comment:
Although It think it could be considered in later pr, but the default fo stop_cb is return false. But how about changing it as nullptr and if the stop_cb is not null the call stop_cb so that we can skip the check if there is stop_cb is not set.